backend/app/usecase/shortlink/creator.go

```diff
@@ -75,34 +75,33 @@ func (c CreatorPersist) CreateShortLink(shortLink entity.ShortLink, customAlias
 		return entity.ShortLink{}, ErrMaliciousLongLink(longLink)
 	}
 
-	if !c.isAliasProvided(customAlias) {
-		return c.createShortLinkWithAutoAlias(shortLink, user)
+	if !(shortLink.Alias != "") {
+		autoAlias, err := c.createAutoAlias()
+		if err != nil {
+			// TODO create error type for fail create auto alias?
```

Comment:
@byliuyang Should I create an error type for auto alias fail? Or should I just throw the error up the call stack as-is?

Reply:
This PR is getting pretty big, considering removing this item from scope. Possibly make issue instead?

Reply:
Issue created. https://github.com/short-d/short/issues/950

Will update the TODO comment with issue number.